src/controller/python/py_matter_yamltest_repl_adapter/matter_yamltest_repl_adapter/runner.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# isort: off
+
+from chip import ChipDeviceCtrl  # Needed before chip.FabricAdmin
+import chip.FabricAdmin  # Needed before chip.CertificateAuthority
+
+# isort: on
+
+import chip.CertificateAuthority
+import chip.logging
+import chip.native
+from chip.ChipStack import *
+from chip.yaml.runner import ReplTestRunner
+from matter_yamltests.runner import TestRunner
+
+
+class Runner(TestRunner):
+    def __init__(self, repl_storage_path: str, commission_on_network_dut: bool):
+        self._repl_runner = None
+        self._chip_stack = None
+        self._certificate_authority_manager = None
+        self._repl_storage_path = repl_storage_path
+        self._commission_on_network_dut = commission_on_network_dut
+
+    async def start(self):
+        chip.native.Init()
+        chip.logging.RedirectToPythonLogging()
+        chip_stack = ChipStack(self._repl_storage_path)
+        certificate_authority_manager = chip.CertificateAuthority.CertificateAuthorityManager(
+            chip_stack, chip_stack.GetStorageManager())
+        certificate_authority_manager.LoadAuthoritiesFromStorage()
+
+        commission_device = False
+        if len(certificate_authority_manager.activeCaList) == 0:
+            if self._commission_on_network_dut == False:
+                raise Exception(
+                    'Provided repl storage does not contain certiciate. Without commission_on_network_dut, there is no reachable DUT')
+            ca = certificate_authority_manager.NewCertificateAuthority()
+            ca.NewFabricAdmin(vendorId=0xFFF1, fabricId=1)
+            commission_device = True
+        elif len(certificate_authority_manager.activeCaList[0].adminList) == 0:
+            certificate_authority_manager.activeCaList[0].NewFabricAdmin(
+                vendorId=0xFFF1, fabricId=1)
+
```

Comment:
Maybe you don't want to do an `elif` but remove the `ca.NewFabricAdmin(vendorId=0xFFF1, fabricId=1)` from the `if` and just do an `if ` to avoid repeating `NewFabricAdmin(vendorId=0xFFF1, fabricId=1)` ?

Something like:
```
        if len(certificate_authority_manager.activeCaList) == 0:
             if self._commission_on_network_dut == False:
                 raise Exception(
                     'Provided repl storage does not contain certificate. Without commission_on_network_dut, there is no reachable DUT')
               certificate_authority_manager.NewCertificateAuthority()
               commission_device = True

        if len(certificate_authority_manager.activeCaList[0].adminList) == 0:
            certificate_authority_manager.activeCaList[0].NewFabricAdmin(
                 vendorId=0xFFF1, fabricId=1)
````
```